### ISSUE TITLE
lsblk with parent kernel name

### ISSUE THREAD

**Issue by eworm-de** [karelzak/util-linux#813]
With `lsblk` from util-linux up to version 2.33.2 I used this a lot:
```
$ lsblk -no KNAME /dev/disk/by-label/EFI-SYSTEM
sda2
$ lsblk -no PKNAME /dev/disk/by-label/EFI-SYSTEM
sda
```
This broke with version 2.34:
```
$ lsblk -no KNAME /dev/disk/by-label/EFI-SYSTEM
sda2
$ lsblk -no PKNAME /dev/disk/by-label/EFI-SYSTEM

```
Reading kernel name works, reading parent kernel name returns nothing.
I tried to bisect, but found a lot of build issues. The offending commit is anywhere between 8229ed2f5c9f663cd363c8fac1454ad886776d5a (last known good) and 0bd05f5ee4876ffd13c98acd56c2bff9971f28f1 (first known bad).

**Comment by karelzak** [comment]
Good catch, thanks for your report! Fixed. The bugfix will be available in v2.34.1.

Note, if you want to print only one device (lsblk <device>) than PKNAME is reliable only for partitions, otherwise PKNAME is based on tree, and in case you print only one device no tree is available. The old versions work in this way too. This is important for example for hierarchy of DM devices, etc.

**Comment by eworm-de** [comment]
Just for reference... Fixed with e3bb9bfb76c17b1d05814436ced62c05c4011f48
Thanks a lot!

**Comment by mfoliveira** [comment]
For documentation purposes:

The underlying issue also affected `lsblk -no RM` (check for removable media) for partitions, and commit e3bb9bf fixes it too.

Those should be the only two columns affected -- only `COL_PKNAME` and `COL_RM` depend on the `parent` paramenter of `device_get_data()`, called by `device_to_scols()` (where the fix makes the `parent` variable non-NULL for partitions.)

```
$ cat /sys/block/sdb/removable 
1

Before:

$ ./lsblk -o PKNAME,RM /dev/sdb1 
PKNAME RM
        0

After:

$ ./lsblk -o PKNAME,RM /dev/sdb1 
PKNAME RM
sdb     1
```
